src/lib/djerba/plugins/tar/sample/plugin.py

```diff
@@ -90,33 +90,53 @@ def extract(self, config):
 
         # If purity is <10%, only report as <10% (not exact number)
         purity = float(purity)
-        rounded_purity = round(purity*100, 1)
+        rounded_purity = round(purity * 100, 1)
         if rounded_purity < 10:
             rounded_purity = "<10"
 
-        results =  {
-                constants.ONCOTREE: config[self.identifier][constants.ONCOTREE],
-                constants.KNOWN_VARIANTS : config[self.identifier][constants.KNOWN_VARIANTS],
-                constants.SAMPLE_TYPE : config[self.identifier][constants.SAMPLE_TYPE],
-                constants.CANCER_CONTENT : rounded_purity,
-                constants.RAW_COVERAGE : int(config[self.identifier][constants.RAW_COVERAGE]),
-                constants.UNIQUE_COVERAGE : int(config[self.identifier][constants.COVERAGE_PL]),
-            }
+        results = {
+            constants.ONCOTREE: config[self.identifier][constants.ONCOTREE],
+            constants.KNOWN_VARIANTS: config[self.identifier][constants.KNOWN_VARIANTS],
+            constants.SAMPLE_TYPE: config[self.identifier][constants.SAMPLE_TYPE],
+            constants.CANCER_CONTENT: rounded_purity,
+            constants.RAW_COVERAGE: int(config[self.identifier][constants.RAW_COVERAGE]),
+            constants.UNIQUE_COVERAGE: int(config[self.identifier][constants.COVERAGE_PL]),
+        }
         data['results'] = results
         return data
 
     def fetch_coverage_etl_data(self, group_id):
         etl_cache = QCETLCache(self.QCETL_CACHE)
         cached_coverages = etl_cache.hsmetrics.metrics
         columns_of_interest = gsiqcetl.column.HsMetricsColumn
-        data = cached_coverages.loc[ (cached_coverages[columns_of_interest.GroupID] == group_id),  [columns_of_interest.GroupID, columns_of_interest.MeanBaitCoverage] ]
+
+        # Filter data for the group_id
+        data = cached_coverages.loc[
+            (cached_coverages[columns_of_interest.GroupID] == group_id),
+            [
+                columns_of_interest.GroupID,
+                columns_of_interest.MeanBaitCoverage,
+                columns_of_interest.TissueType,
+            ]
+        ]
+
         qc_dict = {}
         if len(data) > 0:
-           qc_dict[constants.RAW_COVERAGE] = int(round(data.iloc[0][columns_of_interest.MeanBaitCoverage].item(),0))
+            # Exclude the reference
+            filtered_data = data[data[columns_of_interest.TissueType] != 'R']
+
+            if len(filtered_data) > 0:
+                # Take the first row of the filtered data
```

Comment:
Instead of taking the first row, I think we should check the coverage value is unique. Something like:
```
coverage = filtered_data[columns_of_interest.MeanBaitCoverage].unique()
if len(coverage) != 1:
    # raise an error, multiple coverage values found
else:
    selected_value = coverage[0]